### ISSUE TITLE
custom gcc location, error: redefinition of a 'extern inline' function 'floor_log2' is not supported in C99 mode 

### ISSUE THREAD

**Issue by SeanCode** [crosstool-ng/crosstool-ng#974]
Thanks firstly, I am trying to compile a crosstool with accent gcc version 4.4.6, so I set gcc with source of gcc with custom location under menuconfig panel, only to get this error message, help please! or is there any other way to build such an accent gcc tool?

- Report crosstool-NG version 1.23.0-404-gab3c204a
- Report host OS and its version Darwin 17.4.0
- Attach `.config` (unless the issue is not related to any configuration)
[config.txt](https://github.com/crosstool-ng/crosstool-ng/files/2038610/config.txt)

- Attach `build.log` (unless crosstool-NG fails before the build starts)
[build.log.zip](https://github.com/crosstool-ng/crosstool-ng/files/2038611/build.log.zip)

[INFO ]  Installing pass-1 core C gcc compiler
[EXTRA]    Configuring core C gcc compiler
[EXTRA]    Building gcc
[ERROR]    /Volumes/ct-ng-test/.build/arm-temp-linux-gnueabi/src/gcc/gcc/toplev.c:536:1: error: redefinition of a 'extern inline' function 'floor_log2' is not supported in C99 mode
[ERROR]    /Volumes/ct-ng-test/.build/arm-temp-linux-gnueabi/src/gcc/gcc/toplev.c:571:1: error: redefinition of a 'extern inline' function 'exact_log2' is not supported in C99 mode
[ERROR]    make[2]: *** [Makefile:2470: toplev.o] Error 1
[ERROR]    make[2]: *** Waiting for unfinished jobs....
[ERROR]    make[1]: *** [Makefile:4862: all-gcc] Error 2
[ERROR]
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Installing pass-1 core C gcc compiler'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@373]
[ERROR]  >>        called from: do_gcc_core_backend[scripts/build/cc/gcc.sh@667]
[ERROR]  >>        called from: do_cc_core_pass_1[scripts/build/cc/gcc.sh@207]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@679]
[ERROR]  >>

**Comment by stilor** [comment]
Your build.log shows a successful compilation of GCC 4.8.5, not 4.4.6 as the description claims.
With custom location (and versions older than the oldest supported one), you don't have the benefit of the backported patches in the crosstool-NG. The sources in the custom location are expected to contain all the necessary patches, as desired by the integrator; the version selectors in this case only affect the build steps needed (e.g., different versions of a package may need different configure arguments, etc).

In other words, if you want 4.4.6 to be compilable by a modern host compiler - you'd have to backport the fixes yourself.
